net/kernel_socket.cpp

```diff
@@ -981,19 +990,32 @@ class ETKernelSocketServer : public KernelSocketServer, public NotifyContext {
 /* ET Socket - End */
 
 LogBuffer& operator<<(LogBuffer& log, const IPAddr addr) {
-    return log.printf(addr.d, '.', addr.c, '.', addr.b, '.', addr.a);
+    if (addr.is_ipv4())
+        return log.printf(addr.a, '.', addr.b, '.', addr.c, '.', addr.d);
+    else {
+        return log.printf(addr.to_string());
+    }
```

Comment:
ipv6 addr may has the `::` form. Even we could print every u32 number by order, but using the `ntop` is much simpler.